### ISSUE TITLE
☁️ Migrate Front Door to Cloudflare Part 1/2 - ssw.com.au

### ISSUE THREAD

**Issue by amankumarrr** [SSWConsulting/SSW.Website#2714]
As per my conversation with @wicksipedia and @chrisschultzssw, we are going to place Cloudflare in front of Front Door. We will be testing this on the tfs365.com domain and, upon successful testing, will roll it out to production. This approach aims to reduce downtime during the DNS cutover and ensure a smooth migration from Front Door to Cloudflare.

### Tasks:
 - [x] Staging - Serve traffic from Cloudflare to Front Door.
 - [x] Serve traffic from Cloudflare to Front Door.
 - [x] Move routes to Cloudflare (e.g., /people and /rules).
 - [x] Point Cloudflare to App Service.
 - [x] Add a rule 
 - [x] Send an email for feedback on Front Door vs Cloudflare to Adam

**Comment by amankumarrr** [comment]
@wicksipedia @SSWConsulting/ssw-website-maintainers 

@chrisschultzssw and I have successfully placed Cloudflare in front of Front Door for our staging domain tfs365.com. We've observed a significant improvement in page speeds, thanks to Cloudflare's caching capabilities.

Here are the comparisons:



![Image](https://github.com/SSWConsulting/SSW.Website/assets/71385247/73c587c4-148f-46b3-b544-934fe7ffe66c)

**Figure: Page speed insights for mobiles** 



![Image](https://github.com/SSWConsulting/SSW.Website/assets/71385247/850494ac-63e0-4e86-9d9f-343948a685b0)

**Figure: Page speed insights for mobiles**

**Comment by amankumarrr** [comment]
@wicksipedia @SSWConsulting/ssw-website-maintainers 

**Impacted Routes:**
- https://tfs365.com/
- https://tfs365.com/rules (will be cached)
- https://tfs365.com/people (will be cached)


**Unimpacted Domains from Front Door:**
- https://staging.sswdory.com/
- https://staging.ssw.com/

⚠️ Note: Once we place the `/ssw` route on Cloudflare as a trigger for our worker, v1 will not be accessible afterwards. Ideally, we should do this once we are confident enough to turn off the VM.

**Comment by amankumarrr** [comment]
@SSWConsulting/ssw-website-maintainers 

I'm currently blocked on completing the remaining redirects since @chrisschultzssw needs to plan the migration for ssw.com.au. According to my conversation with him this morning, he anticipates being able to handle this planning and migration later this week.

**Comment by amankumarrr** [comment]
@wicksipedia @bradystroud 

We are currently blocked. As discussed with @chrisschultzssw , he needs to get @ozdemir-mehmet’s approval on the migration plan. If he’s satisfied with it, we will likely proceed with the migration on Monday.

**Comment by amankumarrr** [comment]
I have created a rule for the redirects - https://www.ssw.com.au/rules/handle-redirects/ and modified this rule for the CDN - https://www.ssw.com.au/rules/do-you-use-cdn-for-js-files/

**Comment by amankumarrr** [comment]
@chrisschultzssw 
 
I have created a new issue for the remaining domains - https://github.com/SSWConsulting/SSW.Website/issues/2781

**Comment by Calinator444** [comment]
@wicksipedia 

I spoke with Aman and he told me we couldn't complete this step until we have the /archive redirects in place (e.g. stage 1 of the migration)

> Point App Service to Cloudflare.

**Comment by Calinator444** [comment]
> Send an email for feedback on Front Door vs Cloudflare to Adam


I have an email Drafted for this. I've had it checked by brady but still need it checked by the product owner.

**Comment by Calinator444** [comment]
@bradystroud wanted me to update the rules with information about why we switched to Cloudflare rather than include the information in the  email . I've updated the rule about adding redirects. Still need to update the rule about using CDNs.

https://github.com/SSWConsulting/SSW.Rules.Content/pull/9077

**Comment by Calinator444** [comment]
@wicksipedia 

Done:
- the CName record for ssw.com.au is in Cloudflare now 
- `/rules` and `/people` are being served from a Cloudflare worker (n both tfs365.com and ssw.com)
- Added 2 new rules for handing redirects
  - https://www.ssw.com.au/rules/use-a-cdn/
  - https://www.ssw.com.au/rules/handle-redirects/
